<HTML>
	<HEAD>
	<TITLE>BaRC datasets on tak</TITLE>
	</HEAD>
	<BODY>

	<H3>Bioinformatics datasets available on tak</H3>

	<H4>Datasets are accessible by dataset name in /nfs/BaRC_datasets<BR>
	Each dataset is more thoroughly described in its README file.<BR>
	Datasets may be freely shared unless described as "Whitehead access only".<BR>
	Contact BaRC (bioinformatics@wi.mit.edu) if you have any suggestions of other good resources or would like help with downloading or storage of large datasets that may be of general use.
	</H4>

	<TABLE BGCOLOR="wheat" BORDER=1>
	<TR BGCOLOR="yellow"><TD><B><B>Dataset</TD><TD><B>Description</TD><TD><B>Source</TD></TR><TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/1000_Genomes_PhaseI/README">1000_Genomes_PhaseI</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/1000_YeastStrains/README">1000_YeastStrains</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B>Biobase_other_databases</TD><TD>TRANSCompel and TRANSPRO</TD><TD>Biobase [Whitehead access only]</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/BioCyc_MetaCyc_EcoCyc/README.txt">BioCyc_MetaCyc_EcoCyc</A></TD><TD>Pathways</TD><TD>BioCyc [Whitehead access only]</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/BioGPS/README.txt">BioGPS</A></TD><TD>Normal tissues expression data (array)</TD><TD>BioGPS (GSE1133)</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/Broad_CCLE_Array/README.txt">Broad_CCLE_Array</A></TD><TD>Cancer expression (array)</TD><TD>CCLE</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/Broad_CCLE_RNASeq/README.txt">Broad_CCLE_RNASeq</A></TD><TD>Cancer expression (seq)</TD><TD>CCLE [Whitehead access only]</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/Broad_ExAC/README">Broad_ExAC</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/Broad_GTEx/README.txt">Broad_GTEx</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/cBioPortal_MutationData/README">cBioPortal_MutationData</A></TD><TD>Mutation data</TD><TD>cBioPortal</TD></TR>
<TR><TD><B>CEAS</TD><TD>Data for CEAS plotting tool</TD><TD>CEAS: http://liulab.dfci.harvard.edu/CEAS/</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/COSMIC_Mutations/README">COSMIC_Mutations</A></TD><TD>Cancer genes and mutations</TD><TD>COSMIC</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/CRISPR_Broad_Jan2016/README.txt">CRISPR_Broad_Jan2016</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B>CSHL_LongRnaSeq</TD><TD>ENCODE RNA-seq (selected samples)</TD><TD>UCSC Bioinformatics</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/DiscovEHR_Variants/README.txt">DiscovEHR_Variants</A></TD><TD> </TD><TD> </TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/ENCODE_TFBS/README">ENCODE_TFBS</A></TD><TD> </TD><TD> </TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/Enhancer_Atlas/README.txt">Enhancer_Atlas</A></TD><TD> </TD><TD> </TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/expO/README">expO</A></TD><TD>Tumor expression (array)</TD><TD>expO (GSE2109)</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/FANTOM5_enhancers/README">FANTOM5_enhancers</A></TD><TD>Annotated enhancers</TD><TD>FANTOM</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/fly_modEncode_geneExpr/readme.txt">fly_modEncode_geneExpr</A></TD><TD>Dmel modENCODE RNA-Seq</TD><TD>Flybase</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/Gasch_yeast_respond2enviroment_array/README.txt">Gasch_yeast_respond2enviroment_array</A></TD><TD>Yeast responses to environmental changes (array)</TD><TD>Gasch et al., 2000</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/GSK_Cancer_Cell_Line/README.txt">GSK_Cancer_Cell_Line</A></TD><TD>Cancer cell line expression (array)</TD><TD>GSK</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/HUGO_HGNC_Data/README">HUGO_HGNC_Data</A></TD><TD>HUGO/HGNC annotation</TD><TD>HUGO/HGNC</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/Human_Body_Index_Normal/README.txt">Human_Body_Index_Normal</A></TD><TD>Normal tissue expression (array)</TD><TD>HBI</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/Human_HousekeepingGenes/README.txt">Human_HousekeepingGenes</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/Illumina_Human_Body_Map_2.0/README.txt">Illumina_Human_Body_Map_2.0</A></TD><TD>Normal tissue expression (seq)</TD><TD>Illumina</TD></TR>
<TR><TD><B>Illumina_PlatinumGenomes</TD><TD></TD><TD></TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/InterPro_AllProteinDomains/README">InterPro_AllProteinDomains</A></TD><TD>Annotated protein domains</TD><TD>InterPro</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/KEGG_Pathways/README">KEGG_Pathways</A></TD><TD>Pathways</TD><TD>KEGG</TD></TR>
<TR><TD><B>MEME_matrix_databases</TD><TD></TD><TD></TD></TR>
<TR><TD><B>miRNA_expression_compendia</TD><TD>miRNA expression profiles in human and mouse</TD><TD>microRNA.org</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/NCI60_Cell_Lines/README.txt">NCI60_Cell_Lines</A></TD><TD>Cell lines expression (array)</TD><TD>NCI60</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/ngsplot/README">ngsplot</A></TD><TD>Genome files for ngsplot</TD><TD>ngsplot on Google Drive</TD></TR>
<TR><TD><B>OMIM</TD><TD>Online Mendelian Inheritance in Man</TD><TD>[BaRC access only; cannot be transferred] Downloads</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/ProteinAtlas_RNASeq/README">ProteinAtlas_RNASeq</A></TD><TD>Human Protein Atlas</TD><TD>Human Protein Atlas Downloadable Data</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/pubtator/README.txt">pubtator</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/RepeatMasker/README">RepeatMasker</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/Sanger_Yeast_Resequencing/README">Sanger_Yeast_Resequencing</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B>snpeff_share</TD><TD></TD><TD></TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/TCGA_Array/README">TCGA_Array</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/TCGA_Met450k_IlluminaBeadChip/README">TCGA_Met450k_IlluminaBeadChip</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/TCGA_RNASeqV2/README">TCGA_RNASeqV2</A></TD><TD></TD><TD></TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/TCGA_Tumor_v_Normal_MatchedArray/README">TCGA_Tumor_v_Normal_MatchedArray</A></TD><TD>Matched expression (array)</TD><TD>TCGA</TD></TR>
<TR><TD><B>TCGA_Tumor_v_Normal_MatchedRNASeqV2</TD><TD>Matched expression (seq)</TD><TD>TCGA</TD></TR>
<TR><TD><B>Transfac</TD><TD>Transfac flat files</TD><TD>Biobase [Whitehead access only]</TD></TR>
<TR><TD><B><A HREF="http://tak.wi.mit.edu/barc_datasets/VISTA_enhancers/README">VISTA_enhancers</A></TD><TD>validated human and mouse noncoding fragments</TD><TD></TD></TR>
<TR><TD><B>Yeast_deletion_mutants</TD><TD>expression in yeast deletion mutants</TD><TD>Kemmeren et al., 2014</TD></TR>
<TR><TD><B>Z-Seq_Broad_zebrafish</TD><TD>Zebrafish embryonic expression assay</TD><TD>Broad Institute</TD></TR>
</TD></TR></TABLE>

	</BODY>
	</HTML>
	